### PR TITLE
Adding topics details page

### LINKED ISSUE #366: Add topic details page
we have a tables details page, need to create one for topic as well

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
Closes #366 
### Describe your changes :

I worked on the topics details page because the need to add a topics details page where the user can see the whole detail of the topic.

#
### Type of change :

- [x] New feature

#
### Frontend Preview (Screenshots) :
image

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own. 
- [x] I have tagged my reviewers below.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@shahsank3t, @darth-coder00,

### CODE REVIEW

**Review by shahsank3t — APPROVED** [review]

File: catalog-rest-service/src/main/resources/ui/src/axiosAPIs/topicsAPI.ts [hunk 1]
```diff
@@ -16,6 +16,7 @@
 */
 
 import { AxiosResponse } from 'axios';
+import { Topic } from 'Models';
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/axiosAPIs/topicsAPI.ts:19]: Import this from Topics under `generated` folder

File: catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx [hunk 2]
```diff
@@ -81,47 +96,90 @@ const EntityPageInfo = ({
...
+              <i className="fas fa-tags tw-px-1 tw-mt-2 tw-text-grey-muted" />
+            )}
+            {tier && (
+              <Tags className="tw-bg-gray-200" tag={`#${tier.split('.')[1]}`} />
+            )}
+            {tags.length > 0 && (
+              <>
+                {tags.slice(0, LIST_SIZE).map((tag, index) => (
+                  <Tags
+                    className="tw-bg-gray-200"
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:112]: Why using tailwind color scheme for BG color?

File: catalog-rest-service/src/main/resources/ui/src/axiosAPIs/topicsAPI.ts (hunk 1 above)
Comment (Sachin-chaurasiya) [reply catalog-rest-service/src/main/resources/ui/src/axiosAPIs/topicsAPI.ts:19]: For this PR we can use from models and we can change it in another Separate PR.

File: catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx [hunk 3]
```diff
@@ -81,47 +96,90 @@ const EntityPageInfo = ({
...
+                            key={index}
+                            tag={`#${tag.tagFQN}`}
+                          />
+                        ))}
+                      </>
+                    }
+                    position="bottom"
+                    theme="light"
+                    trigger="click">
+                    <span className="tw-cursor-pointer tw-text-xs link-text v-align-sub tw--ml-1">
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:134]: `link-text` adds underline to `...` and we do not have that anywhere else, we have many either of the behaviour consistent

File: catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx [hunk 4]
```diff
@@ -81,47 +96,90 @@ const EntityPageInfo = ({
...
+                    icon="icon-edit"
+                    title="Edit"
+                    width="12px"
+                  />
+                </button>
+              ) : (
+                <span className="">
+                  <Tags
+                    className="tw-border-main tw-text-primary"
+                    tag="+ Add new tag"
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:175]: Changes `Add new tag` to `Add tag` as it gives wrong impression of being able to add custom tags on the fly

File: catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx (hunk 3 above)
Comment (Sachin-chaurasiya) [reply catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:134]: we do the same thing for column level joins as well.

File: catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx (hunk 4 above)
Comment (darth-coder00) [reply catalog-rest-service/src/main/resources/ui/src/components/common/entityPageInfo/EntityPageInfo.tsx:175]: Make this change everywhere, either in this PR or make a separate PR for it

File: catalog-rest-service/src/main/resources/ui/src/components/schema-editor/SchemaEditor.utils.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { getJSONFromString } from '../../utils/StringsUtils';
+
+export const JSON_TAB_SIZE = 2;
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/components/schema-editor/SchemaEditor.utils.ts:3]: Move this to constants file

File: catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts [hunk 6]
```diff
@@ -322,4 +323,51 @@ declare module 'Models' {
     columns: Array<string>;
     rows: Array<Array<string>>;
   };
+
+  // topic interface start
+  export interface Topic {
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts:328]: No need to add this interface in types.d.ts, use it directly from generated types

File: catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts [hunk 7]
```diff
@@ -322,4 +323,51 @@ declare module 'Models' {
...
+
+  export interface Follower {
+    description: string;
+    href: string;
+    id: string;
+    name: string;
+    type: string;
+  }
+
+  export interface Owner {
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts:356]: We already have Owner type, why do we need interface as well?

File: catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts (hunk 6 above)
Comment (Sachin-chaurasiya) [reply catalog-rest-service/src/main/resources/ui/src/interface/types.d.ts:328]: okay.

File: catalog-rest-service/src/main/resources/ui/src/pages/topic-details/index.tsx [hunk 8]
```diff
@@ -0,0 +1,333 @@
+import { AxiosResponse } from 'axios';
+import { compare } from 'fast-json-patch';
+import { observer } from 'mobx-react';
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/pages/topic-details/index.tsx:3]: Why we need observer if we do not have appstate in this component?

File: catalog-rest-service/src/main/resources/ui/src/pages/topic-details/index.tsx [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+import { ColumnTags, TableDetail, Topic } from 'Models';
```
Comment (darth-coder00) [inline catalog-rest-service/src/main/resources/ui/src/pages/topic-details/index.tsx:4]: Topic from `generated` folder

**Review by darth-coder00 — CHANGES_REQUESTED** [review]
Requested few changes

**Comment by darth-coder00** [comment]
Also add some max height to the container above `Codemirror`, in case schema is too long, scroll should come in that container after a certain height

**Comment by Sachin-chaurasiya** [comment]
Addressed every comment except that use types from generated types.

**Review by darth-coder00 — APPROVED** [review]

**Pull request merged**